flink-clients/src/main/java/org/apache/flink/client/deployment/application/FromJarEntryClassInformationProvider.java

```diff
@@ -68,11 +68,33 @@ private FromJarEntryClassInformationProvider(
         this.jobClassName = jobClassName;
     }
 
+    /**
+     * Returns the specified {@code jarFile}. If no {@code jarFile} was specified it can only return
+     * an empty {@code Optional} if the {@code jobClassName} is specified instead. In that case, the
```

Comment:
The condition when it returns an empty optional is unnecessary and just creates confusion. The jar _and_ classname not being specified is a case that a consumer of this APIn ever has to be concerned with because the constructor rejects it.

Reply:
Fair enough, the information is also given in `@return` docs. So, I guess, that's good enough.